### ISSUE TITLE
Norm estimation missing (?)

### ISSUE THREAD

**Issue by mfasi** [JuliaLang/julia#12267]
As far as I could see, there are no Julia functions that perform 1, 2 and inf norm estimation (neither for sparse nor for dense matrices). Provided that the functionality is actually missing and it isn't just me who couldn't find it, is there some reason why it hasn't been implemented yet? If that's not the case, I'd be happy to work on the issue.

**Comment by IainNZ** [comment]
Forgive my ignorance, but what is norm estimation for a matrix?

**Comment by pao** [comment]
I'm not sure either, but this tag seems safe.

**Comment by IainNZ** [comment]
This? http://www4.ncsu.edu/~ipsen/ps/slides_ima.pdf

**Comment by IainNZ** [comment]
@mfasi I don't think we have this in Base, but if you'd be interested in implementing it then I'd suggest making a package first, and the discussion about moving it into Base or not can come at a later date.

**Comment by ViralBShah** [comment]
Yes, this seems like something to start out in a package.

**Comment by jiahao** [comment]
I've already implemented two different randomized norm estimators [in IterativeSolvers.jl](https://github.com/JuliaLang/IterativeSolvers.jl/blob/3e3491acad023f03164b465b06243f4de4b68e72/src/rlinalg.jl#L49-L141). You're more than welcome to contribute other estimates.

See also #4767

**Comment by mfasi** [comment]
I'm sorry, it looks like it was impossible to understand what I wanted to do. I reworded the title and the first post using the correct vocabulary and will provide a few details here. I wanted to implement something much like the functions `normest` and `normest1` of Matlab. I guess that these functions are mainly used by people dealing with sparse matrices, but they arise when dealing with dense matrices as well. I think that having these two functions would contribute to fix #6485 and also to implement some of the algorithms in #5840.

Sorry for the misunderstanding, I hope I explained myself now.
